scripts/utils/ensure_packages_installed.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# Function to ensure necessary packages are installed
+ensure_homebrew_packages_installed() {
+    # Check if Homebrew is installed
+    if ! command -v brew >/dev/null 2>&1; then
+        echo "Homebrew is not installed on this system."
+        echo "Please install via https://brew.sh/ before rerunning this script."
+        return 1
+    fi
+
+    # Iterate over all arguments which are package names
+    for package_name in "$@"; do
+        if brew list --formula | grep -q "^${package_name}\$"; then
+            echo "${package_name} is already installed."
+        else
+            echo "${package_name} is not installed. Installing ${package_name}..."
+            brew install "${package_name}"
```

Comment:
Should we install these packages within the venv as well, along with pyspelling?

Reply:
They're system installs, so I'm not sure we should here.

Reply:
Is this step still breaking for you if a package isn't installed?

Reply:
Not even for system installs that are (for our purposely probably) only going to be used with the spellchecker e.g. `aspell`, which `pyspelling` is built on top of?

Reply:
Nope, installs are working fine now 🙌

Reply:
If we're using homebrew to install these, I'm not sure we can.

If we could, we'd need to make an additional os environment as these can't be installed in the python venv - I'm not sure it's worth the faff of setting up (or if it's possible outside of docker).